tracing-attributes/src/lib.rs

```diff
@@ -524,22 +527,46 @@ fn gen_block(
                             Err(e)
                         }
                     }
-                }, __tracing_attr_span).await
+                };
+                if tracing::level_enabled!(#level) {
+                    tracing::Instrument::instrument(
+                        fut,
+                        __tracing_attr_span
+                    )
+                    .await
```

Comment:
in this case, since we _are_ always creating the span, we could actually do this:
```suggestion
                if !span.is_disabled() {
                    tracing::Instrument::instrument(
                        fut,
                        __tracing_attr_span
                    )
                    .await
```
which would also let us skip instrumenting the future in the case where the span's _level_ is enabled but it was disabled by the subscriber (e.g. if its target was not enabled)